### PR TITLE
Keep a stack of frames on the VM.

### PR DESCRIPTION
**Pull request by cthulahoops**

### CODE REVIEW

File: vm/src/vm.rs [hunk 1]
```diff
@@ -52,13 +53,28 @@ impl VirtualMachine {
...
+        match self.run_frame(frame)? {
+            ExecutionResult::Return(value) => Ok(value),
+            _ => panic!("Got unexpected result from function"),
+        }
+    }
+
+    pub fn run_frame(&mut self, frame: Frame) -> Result<ExecutionResult, PyObjectRef> {
+        let frame = self.ctx.new_frame(frame);
+        self.frames.push(frame.clone());
+        let mut frame = objframe::get_value(&frame);
```
Comment (windelbouwman) [inline vm/src/vm.rs]: Maybe we can seperate the frame into a part that is immutable (like the codeobject) and a part that is mutable (like the current program counter `lasti`). Or at least give the frame a PyRef to the codeobject.

File: vm/src/frame.rs [hunk 2]
```diff
@@ -39,7 +38,6 @@ enum Block {
     },
 }
 
-#[derive(Clone)]
```
Comment (windelbouwman) [inline vm/src/frame.rs]: Ow yeah!

Comment (cthulahoops) [reply vm/src/frame.rs]: It doesn't work though.

I'll guess I'll have to split the frame up someone so we don't have to keep mutable references to it around.

Comment (windelbouwman) [reply vm/src/frame.rs]: Yes, I believe so too, and we should probably use more of the PyObject and PyFrame types, which are references (`Rc<RefCell<>>`), and we can borrow them mutably or immutably.

**Pull request closed without merging**